### ISSUE TITLE
Tensorflow Model Compiler Metrics Error: NaN Outputs

### ISSUE THREAD

**Issue by abojanich** [tensorflow/tensorflow#55277]
I am having a problem using Tensorflow when trying to use two custom metric functions which calculate the mean and maximum fractional errors. My code for these functions are the following:

```
meanFE = np.sum(np.abs(f_emulator - f_test))
maxFE = np.max(np.abs(f_emulator - f_test))
```

When I try to use these metrics, all of the loss outputs go to NaN. If I use my meanFE and maxFE outside of Tensorflow, I don't encounter this issue. The only way I have been able to use these functions in Tensorflow is to remove the denominator term in the fractional errors, but this defeats the purpose of using a fractional error in the first place. I suspect this has something to do with the fact that my arrays get turned into Tensor objects by default when the system trains. My inputs are astronomical spectra (i.e., each spectrum is an array of shape (500 x 100), or there are 500 spectral arrays with 100 wavelength bins per array.

I have tried using other built in functions for my metrics, namely the MAPE function (mean absolute percentage error), but this generates bad spectra (output spectra do not look like input spectra), so I am trying to implement my own shown above. Has anyone encountered NaN outputs as your loss and metric output when using functions such as these? I am using Tensorflow 1.14.0 on macOS.

**Comment by mohantym** [comment]
Hi @abojanich ! 1.x versions are not supported anymore. I think there might be NaN values in Dataset itself as you are already using np.abs to convert the negative fraction values to positive ones.  Please let us know after replacing the above metrics with the below one.  If it is a zero division error in your loss function , you can resolve it using adding [k.epsilon](https://stackoverflow.com/a/69679222) in Denominator.  Please post on [TF forum ](https://discuss.tensorflow.org/)for further assistance. Thank you!
```
meanFE = np.abs(np.sum(np.abs(np.nan_to_num(f_emulator - f_test)))) // I think np.mean instead of np.sum for getting the mean 
maxFE = np.abs(np.max(np.abs(np.nan_to_num(f_emulator - f_test))))
```
